lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextManager.cs

```diff
@@ -338,8 +339,9 @@ private void StartTask(StartTaskProto startTaskProto)
                         string.Format(CultureInfo.InvariantCulture, "Task expected context '{0}' but the active context has Id '{1}'", expectedContextId, currentActiveContext.Id));
                     Utilities.Diagnostics.Exceptions.Throw(e, LOGGER);
                 }
-                TaskConfiguration taskConfiguration = new TaskConfiguration(startTaskProto.configuration);
-                currentActiveContext.StartTask(taskConfiguration, expectedContextId, _heartBeatManager);
+                
+                var configuration = new AvroConfigurationSerializer().FromString(startTaskProto.configuration);
```

Comment:
AvroConfigurationSerializer can be injected in the constructor of this class without any additional settings.